### DISCUSSION TITLE
`helpers:pinGitHubActionDigestsToSemver` not working with `pin-dependencies`

### DISCUSSION THREAD

**Discussion by MPV** [renovatebot/renovate#27194] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

I can confirm that `helpers:pinGitHubActionDigestsToSemver` works really well for upgrading versions to semver. 🎉 

However, I may have run into a follow-up issue. Doee anyone have ideas why the "pin digests" functionality doesn't combine with `helpers:pinGitHubActionDigestsToSemver`? 

See minimal reproduction repo and example here:
- https://github.com/MPV/renovate-pinGitHubActionDigestsToSemver-should-pinDigests-to-semver/pull/2

For example, here's a sibling PR showing the preset working in normal cases:
- https://github.com/MPV/renovate-pinGitHubActionDigestsToSemver-should-pinDigests-to-semver/pull/3

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by HonkingGoose** [comment]
@TWiStErRob do you have any idea what's going wrong here? I remember you being very helpful when I created the `helpers:pinGitHubActionDigestsToSemver` preset. So maybe you want to help again? 😄

**Reply by MaxymVlasov** [reply]
I saw such issues while using a workaround that became the foundation for `helper`.

```json5
{
  "extends": ["helpers:pinGitHubActionDigests"],
  "packageRules": [{
      "description": "v prefix workaround for action updates",
      "matchDepTypes": ["action"],
      "extractVersion": "^(?<version>v\\d+\\.\\d+\\.\\d+)$",
      "versioning": "regex:^v(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?$"
    }]
}
```

During that time, most of them were resolved during the next update windows, so it's kinda "self-healing bug". No idea why it can't work well on the first try.

**Reply by MPV** [reply]
FYI: If anyone looks this week, there's another issue currently at play with the reproduction repo, which is described in and hopefully fixed by:
- https://github.com/renovatebot/renovate/pull/27453

**Reply by TWiStErRob** [reply]
Sorry, I just saw this, the inner workings of Renovate are very advanced in this. I didn't understand the logic even on the original issue :) Optional v could be it, or anything else :) I haven't started using this yet to see how it works in practice.

@MaxymVlasov, a heads up: you pull in the built-in preset and your custom one. They might both match?

**Reply by MaxymVlasov** [reply]
@TWiStErRob Yeah, they should both match. But I'm not sure how `packageRules` rules apply to presets

**Reply by MPV** [reply]
We've "slept on it" for a year, anyone that happens to have any new/better views on it at this point? 🙏
